File: android/src/main/kotlin/io/jimmytai/flutter_tencent_vap/FlutterTencentVapUtils.kt
package io.jimmytai.flutter_tencent_vap

import com.tencent.qgame.animplayer.util.ScaleType

object FlutterTencentVapUtils {

    fun parseVapViewAlignment(alignment: String): VapViewAlignment {
        return when (alignment) {
            "top" -> VapViewAlignment.TOP
            "center" -> VapViewAlignment.CENTER
            "bottom" -> VapViewAlignment.BOTTOM
            else -> VapViewAlignment.CENTER
        }
    }

    fun parseContentMode(contentMode: String): TencentVapContentMode {
        return when (contentMode) {
            "fill" -> TencentVapContentMode.Fill
            "contain" -> TencentVapContentMode.Contain
            "cover" -> TencentVapContentMode.Cover
            else -> TencentVapContentMode.Cover
        }
    }

    fun convertContentModeToVap(contentMode: TencentVapContentMode): ScaleType {
        return when(contentMode) {
            TencentVapContentMode.Fill -> ScaleType.FIT_XY
            TencentVapContentMode.Contain -> ScaleType.FIT_CENTER
            TencentVapContentMode.Cover -> ScaleType.CENTER_CROP
        }
    }
}

File: ios/Classes/FlutterTencentVapUtils.swift

class FlutterTencentVapUtils {
  public static func parseAlignment(alignment: String) -> TencentVapAlignment {
    switch alignment {
    case "top":
      return .top
    case "center":
      return .center
    case "bottom":
      return .bottom
    default:
      return .center
    }
  }
  
  public static func parseContentMode(contentMode: String) -> TencentVapContentMode {
    switch contentMode {
    case "fill":
      return .fill
    case "contain":
      return .contain
    case "cover":
      return .cover
    default:
      return .cover
    }
  }
}
